codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -193,6 +193,7 @@ public ChannelFuture writePriority(ChannelHandlerContext ctx, int streamId, int
             // The set priority operation must be done before sending the frame. The parent may not yet exist
             // and the priority tree may also be modified before sending.
             stream.setPriority(streamDependency, weight, exclusive);
+            stream.frameSent();
```

Comment:
Yep, this was an oversight. I think this goes away by switching to `isHeaderSent`.